pilot/pkg/model/config.go

```diff
@@ -277,48 +277,46 @@ func resolveGatewayName(gwname string, meta config.Meta) string {
 	return out
 }
 
-// MostSpecificHostMatch compares the map of the stack to the needle, and returns the longest element
-// matching the needle, or false if no element in the map matches the needle.
-func MostSpecificHostMatch[V any](needle host.Name, m map[host.Name]V) (host.Name, bool) {
-	matches := []host.Name{}
-	// exact match first
-	if m != nil {
-		if _, ok := m[needle]; ok {
-			return needle, true
+// MostSpecificHostMatch compares the maps of specific and wildcard hosts to the needle, and returns the longest element
+// matching the needle and it's value, or false if no element in the maps matches the needle.
+func MostSpecificHostMatch[V any](needle host.Name, specific map[host.Name]V, wildcard map[host.Name]V) (host.Name, V, bool) {
```

Comment:
seems host.Name, V are redundant

Reply:
why would they be redundant?

Reply:
some insight would be greatly appreciated

Reply:
As the callers only use one return value. But not a big issue, could be needed in future

Reply:
Callers are using all return values, some are using one, other the other one